test/testdata/infer/generics/arity.rb

```diff
@@ -0,0 +1,40 @@
+# typed: true
```

Comment:
Looks like this test duplicates`test/testdata/infer/generics/arity_mismatch.rb`, can you merge the two?